main.go

```diff
@@ -248,6 +253,29 @@ func (a *autographer) disableDebug() {
 	return
 }
 
+// startCleanupHandler sets up a chan to catch int, kill, term
+// signals and run signer AtExit functions
+func (a *autographer) startCleanupHandler() {
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, os.Interrupt, os.Kill, syscall.SIGTERM)
```

Comment:
would this catch autograph itself panicking?

Reply:
It should just catch the int, kill, and term signals (though Hal pointed out that kill isn't catchable).

If there's a way to panic and trigger those signals I assume they'd be caught, but other signals
like SIGBUS, SIGFPE, and SIGSEGV get converted into runtime panics as usual.

This does mean we'd leave the sec and key rings on disk, but I think they'd be helpful for debugging and don't want to mask actual panics.

https://golang.org/pkg/os/signal/#hdr-Types_of_signals